### DISCUSSION TITLE
Participate to early testing program for the type checker

### DISCUSSION THREAD

**Discussion by millsp** [swc-project/swc#1425] [Ideas]
I see that you are moving forward with the conformance tests, well done! I would like to run tests for when your checker is closer to an alpha release. I have some complex type logic over here at [ts-toolbelt](https://github.com/millsp/ts-toolbelt) that I could challenge it with.

We effectively use all type system features (no code inference, just types). The utility types combine all the available TypeScript features in the most complex ways. It is extensively tested, maybe SWC could lean on some of these tests. I would be up to the task to set such tests up.

Such types are becoming increasingly adopted by the community. Your type checker would definitely bring a new era of performant type-level programming. Your work is very welcome.

**Comment by kdy1** [comment]
@millsp Thank you so much for your interest!
I only considered open beta, but EAP would be definitely useful.
I'll also make an alpha release to harden it. :)

Btw, ts-toolbelt seems very interesting. I didn't know that such operations are possible.
